django/db/models/sql/query.py

```diff
@@ -534,8 +534,13 @@ def exists(self):
                 # SELECT clause which is about to be cleared.
                 q.set_group_by(allow_aliases=False)
             q.clear_select_clause()
+        if q.combined_queries and q.combinator == 'union':
+            q.combined_queries = tuple(
+                combined_query.exists(limit=False) for combined_query in q.combined_queries
```

Comment:
I wonder if we want a database feature here for which backends support limiting within a combined query could be useful.